universum_log_collapser/src/main/resources/io/jenkins/plugins/universum_log_collapser/Note/script.js

```diff
@@ -30,6 +32,34 @@ function finishColoring() {
     setTimeout(colorFailedSections, coloringTimeoutMilliSeconds);
 }
 
+/*
+When Universum is executed using pipeline "sh", pipeline span brokes first section collapse/expand mechanism.
+Finding target <span> element and simply move a content out from tag body.
+*/
+function fix_pipeline() {
+    var spans = document.getElementsByTagName("span");
+    var logStartRegexp = new RegExp("==&gt; Universum \\d+\\.\\d+\\.\\d+ started execution");
+    var classNameRegexp = new RegExp("^pipeline-node-\\d+$");
+    var fixed = false;
+    
+    for (var i = 0; i < spans.length; i++) {
+        var el = spans[i];
+        if ((logStartRegexp.exec(el.innerHTML) == undefined) || (classNameRegexp.exec(el.className) == undefined)) {
+            continue;
+        }
+
+        el.nextSibling.insertAdjacentHTML("beforebegin", el.innerHTML);
+        el.innerHTML = "";
+        fixed = true;
+        break;
+    }
+
+    if (!fixed) {
+        setTimeout(fix_pipeline, 50);
```

Comment:
1. Why you need setTimeout?
2. What is the guarantee it is ever going to finish?

Reply:
Refactored in next commit.